src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,128 @@
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves the list of tasks to the task file.
+     *
+     * @param listOfTasks Array list of tasks.
+     * @throws DukeException If the tasks cannot be saved to the file.
+     */
+    public void save(ArrayList<Task> listOfTasks) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            String str;
+            for (Task task : listOfTasks) {
+                str = task.toString();
+                str = str.replace("[ ]", " | 0 |");
+                str = str.replace("[X]", " | 1 |");
+                str = str.replace("[", "");
+                str = str.replace("]", "");
+                if (str.startsWith("D")) {
+                    str = str.replace("(by:", "|");
+                    str = str.replace(")", "");
+                } else if (str.startsWith("E")) {
+                    str = str.replace("(from:", "|");
+                    str = str.replace(" to: ", " - ");
+                    str = str.replace(")", "");
+                }
+                str = str + "\n";
+                fw.write(str);
+            }
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException(e);
+        }
+    }
+
+    /**
+     * Returns the tasks from the task file.
+     *
+     * @return Array list of tasks.
+     * @throws DukeException If the file could not be found or has incorrect date time format.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> listOfTasks = new ArrayList<>(100);
+        try {
+            File f = new File(filePath);
+            Scanner s = new Scanner(f);
+            Task task;
+            String description, str, taskInfo;
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM dd yyyy h:mm a");
+
+            while (s.hasNext()) {
+                str = s.nextLine();
+                taskInfo = str.substring(8);
+                if (str.startsWith("T")) {
+                    description = taskInfo;
+                    task = new Todo(description);
+                } else if (str.startsWith("D")) {
+                    int byIdx = taskInfo.indexOf("|");
+                    description = taskInfo.substring(0, byIdx - 1);
+                    LocalDateTime by = LocalDateTime.parse(taskInfo.substring(byIdx + 2), format);
+                    task = new Deadline(description, by);
+                } else {
+                    int fromIdx = taskInfo.indexOf("|");
+                    int toIdx = taskInfo.indexOf("-");
+                    description = taskInfo.substring(0, fromIdx - 1);
+                    LocalDateTime from = LocalDateTime.parse(taskInfo.substring(fromIdx + 2, toIdx - 1), format);
+                    LocalDateTime to = LocalDateTime.parse(taskInfo.substring(toIdx + 2), format);
+                    task = new Event(description, from, to);
```

Comment:
Would using Switch Case be better here? In view of future feature expansion where more task types could be introduced.